src/kv/deserialise.h

```diff
@@ -583,22 +626,50 @@ namespace kv
         public_only_,
         v_,
         std::move(changes_),
-        std::move(new_maps_))
+        std::move(new_maps_)),
+      max_conflict_version(max_conflict_version_)
     {
       tx = std::move(tx_);
     }
 
-    ApplyResult execute() override
+    ApplyResult apply() override
     {
-      return fn(tx, term, changes);
+      return fn(tx, term, changes, req);
     }
 
     std::function<ApplyResult(
-      std::unique_ptr<Tx>& tx, Term term, OrderedChanges& changes)>
-      fn = [](std::unique_ptr<Tx>& tx, Term term, OrderedChanges& changes)
-      -> ApplyResult {
+      std::unique_ptr<Tx>& tx,
+      Term term,
+      OrderedChanges& changes,
+      aft::Request& req_)>
+      fn = [](
```

Comment:
Is this a public `std::function` member called `fn`? Why is it implemented like this, can't this simply be inlined into `apply`? Or at least be a method rather than a member?

Reply:
Reading again, I really don't follow this - it operates entirely on its own members. and the real result is that it populates `tx` and `req`. What am I missing?

Reply:
ok so i have been working with this code for too long :(, the artifact is fixed, fn no longer exists :)

So what you are doing in this function is separating 2 steps, step 1 is creating the tx object (from the append entry) and in an asynchronous step running it.